app/src/main/java/com/openclassrooms/realestatemanager/ui/EstateAddFragment.kt

```diff
@@ -0,0 +1,104 @@
+                numbersOfRoomsEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_numbersOfRooms_required), Toast.LENGTH_SHORT).show()
+                numbersOfBathroomsEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_numbersOfBathrooms_required), Toast.LENGTH_SHORT).show()
+                numbersOfBedroomsEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_numbersOfBedrooms_required), Toast.LENGTH_SHORT).show()
+                houseNumberEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_houseNumber_required), Toast.LENGTH_SHORT).show()
+                streetEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_street_required), Toast.LENGTH_SHORT).show()
+                zipCodeEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_zipCode_required), Toast.LENGTH_SHORT).show()
+                cityEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_city_required), Toast.LENGTH_SHORT).show()
+                countryEditText.text.toString().isEmpty() -> Toast.makeText(activity, resources.getString(R.string.add_estate_country_required), Toast.LENGTH_SHORT).show()
+                else -> {
+                    val estate = Estate(
+                        id = System.currentTimeMillis(),
+                        type = type!!,
+                        price = priceEditText.text.toString().toDouble(),
+                        surface = surfaceEditText.text.toString().toFloat(),
+                        numberOfRooms = numbersOfRoomsEditText.text.toString().toInt(),
+                        numberOfBathrooms = numbersOfBathroomsEditText.text.toString().toInt(),
+                        numberOfBedrooms = numbersOfBedroomsEditText.text.toString().toInt(),
+                        description = descriptionEditText.text.toString(),
+                        houseNumber = houseNumberEditText.text.toString().toInt(),
+                        street = streetEditText.text.toString(),
+                        additionalAddress = additionalAddressEditText.text.toString(),
+                        zipCode = zipCodeEditText.text.toString(),
+                        city =  cityEditText.text.toString(),
+                        country = countryEditText.text.toString(),
+                        status = Estate.Status.AVAILABLE,
+                        entryDate = Calendar.getInstance(),
+                        saleDate = null,
+                        agent = "Agent"
+                    )
+                    Toast.makeText(activity, resources.getString(R.string.add_estate_added), Toast.LENGTH_LONG).show()
+                    viewModel.addEstate(estate)
```

Comment:
```suggestion
                    viewModel.addEstate(estate)
                    Toast.makeText(activity, resources.getString(R.string.add_estate_added), Toast.LENGTH_LONG).show()
```
Toujours mieux de feedback une fois l'action réellement faite ;)

Reply:
Dans un projet plus complet, tu seras surement amené à gérer les échecs (avec un Result en retour de fonction par exemple), ça te permet de de l'afficher qu'en cas de succès : 

```kotlin
viewModel.addEstate(estate)
    .onSuccess {
        Toast.makeText(activity, resources.getString(R.string.add_estate_added), Toast.LENGTH_LONG).show()
    }.onFailture { 
        Toast.makeText(activity, resources.getString(R.string.add_estate_addition_failed), Toast.LENGTH_LONG).show()
    }
```

--> Mais je ne pense pas qu'il soit nécessaire de le faire dans le cadre de ce projet.

Reply:
Ok

Reply:
Tu me montera ça en vision sa m'interesse